addons/account/models/account_payment.py

```diff
@@ -638,6 +638,12 @@ def _check_payment_method_line_id(self):
             if not pay.payment_method_line_id:
                 raise ValidationError(_("Please define a payment method line on your payment."))
 
+    @api.constrains('payment_method_line_id', 'journal_id')
+    def _check_payment_method_line_journal(self):
+        for payment in self:
+            if payment.payment_method_line_id.journal_id and payment.payment_method_line_id.journal_id != payment.journal_id:
+                raise ValidationError("The selected payment method is from a different journal, please select the payment method again.")
```

Comment:
this should be translated. Do we really need to duplicate this constraint in the wizard?

Reply:
also, I think we should check `available_payment_method_lines` here, because even if the journal is the same, the payment method might still be invalid (outbound/inbound/filtered/etc)

Reply:
I've adapted the changes :)

Just have 2 questions:
Should I add the constrains of `available_payment_method_line_ids` (`journal_id`, `payment_type`, `currency_id`)?

There's something that will fail now but don't know if that matters:
It is possible to remove a payment method line from a journal, and so the payment method line does not have any `journal_id` anymore. 
So if you reset a to draft a payment that had that method line and re-post it, the validation error will be raised as this method line is no longer in the available ones.
Don't know if that matters, the user can still select a new one.

Reply:
> Do we really need to duplicate this constraint in the wizard?

No, don't know why I did that

Reply:
do you mean that if you reset a payment to draft, then remove that payment method from the journal, and then try to confirm the payment, the error will be raised? If so, in my opinion, that's ok.
there are more tests failing though :slightly_smiling_face:

Reply:
oh, and the translation should be added in the .pot file :slightly_smiling_face:

Reply:
> there are more tests failing though 🙂

The last 2 tests that fails are for payment transactions refunds.
I wonder if it is not a bug in the code:
Here we see that the payment can be inbound/outbound https://github.com/odoo/odoo/blob/799465ec53e0a07890ae6bcf3b0b7b5b46fdc00b/addons/payment/models/payment_transaction.py#L930
But it will only take inbound payment methods: https://github.com/odoo/odoo/blob/799465ec53e0a07890ae6bcf3b0b7b5b46fdc00b/addons/payment/models/payment_transaction.py#L926

Reply:
> The last 2 tests that fails are for payment transactions refunds. I wonder if it is not a bug in the code: Here we see that the payment can be inbound/outbound

hmm it looks like a bug indeed :thinking:

Reply:
I wonder if we could just add a domain on the payment method?

Reply:
hi, could you please let me know if you have an update on this? :slightly_smiling_face: